src/screens/MiningServer.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import { t } from 'ttag';
+import {
+  DEFAULT_MINING_SERVERS,
+} from '../constants';
+import LOCAL_STORE from '../storage';
+import { getGlobalWallet } from "../modules/wallet";
+import BackButton from '../components/BackButton';
+import HathorAlert from '../components/HathorAlert';
+
+/**
+ * Screen to change the tx-mining server that the wallet uses to mine transactions
+ *
+ * @memberof Screens
+ */
+function MiningServer() {
+  const wallet = getGlobalWallet();
+
+  /* errorMessage {string} Message to be shown in case of error in form */
+  const [errorMessage, setErrorMessage] = useState('');
+  /* isHardwareWallet {boolean} If the application is using a hardware wallet */
+  const [isHardwareWallet] = useState(LOCAL_STORE.isHardwareWallet());
+  /* currentServer {string} currently connected tx-mining server url */
+  const [currentServer, setCurrentServer] = useState('');
+
+  const currentNetwork = wallet.getNetwork();
+  const defaultServer = DEFAULT_MINING_SERVERS[currentNetwork]
+
+  useEffect(() => {
+    setCurrentServer(wallet.storage.config.getTxMiningUrl());
```

Comment:
Why do you need this in the `useEffect`? You can just use `wallet.storage.config.getTxMiningUrl()` as the default state

Reply:
This effect does not exist anymore